src/lib/sticky-header/sticky-header.ts

```diff
@@ -0,0 +1,321 @@
+
+  isStuck: boolean = false;
+  /** Whether the browser support CSS sticky positioning. */
+  private _isStickyPositionSupported: boolean = true;
+
+
+  /** The element with the 'cdkStickyHeader' tag. */
+  element: HTMLElement;
+  /** The upper container element with the 'cdkStickyRegion' tag. */
+  stickyParent: HTMLElement | null;
+  /** The upper scrollable container. */
+  upperScrollableContainer: HTMLElement;
+  /**
+   * The original css of the sticky element, used to reset the sticky element
+   * when it is being unstuck
+   */
+  private _originalStyles = {} as CSSStyleDeclaration;
+  /**
+   * 'getBoundingClientRect().top' of CdkStickyRegion of current sticky header.
+   * It is used with '_stickyRegionBottomThreshold' to judge whether the current header
+   * need to be stuck.
+   */
+  private _stickyRegionTop: number;
+  /**
+   * Bottom of the sticky region offset by the height of the sticky header.
+   * Once the sticky header is scrolled to this position it will stay in place
+   * so that it will scroll naturally out of view with the rest of the sticky region.
+   */
+  private _stickyRegionBottomThreshold: number;
+
+  private _onScrollSubscription: Subscription;
+
+  private _onTouchSubscription: Subscription;
+
+  private _onResizeSubscription: Subscription;
+
+  constructor(element: ElementRef,
+              scrollable: Scrollable,
+              @Optional() public parentRegion: CdkStickyRegion,
+              platform: Platform) {
+    if (platform.isBrowser) {
+      this.element = element.nativeElement;
+      this.upperScrollableContainer = scrollable.getElementRef().nativeElement;
+      this.setStrategyAccordingToCompatibility();
+    }
+  }
+
+  ngAfterViewInit(): void {
+    if (!this._isStickyPositionSupported) {
+      this.stickyParent = this.parentRegion != null ?
+        this.parentRegion._elementRef.nativeElement : this.element.parentElement;
+      let values = window.getComputedStyle(this.element, '');
+      this._originalStyles = {
+        position: values.position,
+        top: values.top,
+        right: values.right,
+        left: values.left,
+        bottom: values.bottom,
+        width: values.width,
+        zIndex: values.zIndex} as CSSStyleDeclaration;
+      this.attach();
+      this.defineRestrictionsAndStick();
+    }
+  }
+
+  ngOnDestroy(): void {
+    if (this._onScrollSubscription) {
+      this._onScrollSubscription.unsubscribe();
+    }
+
+    if (this._onResizeSubscription) {
+      this._onResizeSubscription.unsubscribe();
+    }
+
+    if (this._onTouchSubscription) {
+      this._onTouchSubscription.unsubscribe();
+    }
+  }
+
+  /**
+   * Check if current browser supports sticky positioning. If yes, apply
+   * sticky positioning. If not, use the original implementation.
+   */
+  setStrategyAccordingToCompatibility(): void {
+    this._isStickyPositionSupported = isPositionStickySupported();
+    if (this._isStickyPositionSupported == true) {
```

Comment:
can just do `if (this._isStickyPositionSupported)` - shouldn't ever be the case to need `== true` or `== false` in an if conditional. Just use the value